src/arcsf/data/data_utils.py

```diff
@@ -0,0 +1,244 @@
+import itertools
+
+import numpy as np
+from datasets import Dataset, load_dataset
+
+
+def get_indices_structured(
+    forgotten_author_numbers: list,
+    q_remove: int = 4,
+    n_authors: int = 200,
+    q_per_author: int = 20,
+):
+    """
+    Returns two flattened lists of Question--Answer indices given authors numbers to
+    remove/retain, in the case where the forget target can be easily structured.
+    Assumes equal questions per author, and sorted authors.
+
+        Parameters:
+                forgotten_author_numbers (list):    List of integers containing author
+                                                    indexes about which facts should
+                                                    be removed
+                n_authors (int):                    Total number of authors
+                q_removal (int):                    Number of questions/facts that
+                                                    should be removed
+                q_per_author (int):                 Integer defining the number of
+                                                    questions to remove per author
+
+        Returns:
+                forget_indices (list):              List of integers corresponding to
+                                                    the question indices pertaining the
+                                                    authors in author_numbers
+                retain_indices (list):              List of integers corresponding to
+                                                    the remaining question indices
+    """
```

Comment:
Fantastic docstrings throughout - both informative and succinct. A minor comment in that in general we don't type the params/returns in the doc string as these should already be type hinted (see comment above). Also suggest reducing whitespace between param/return name and text, see an example docstring from an ARC project here: https://github.com/alan-turing-institute/ARC-LoCoMoSeT/blob/c0598a4e0aa1d1e4d7326ff3492ff6831b1a63eb/src/locomoset/datasets/preprocess.py#L10

Reply:
The LoCoMoSeT docstring style is from the Google Python styleguide fyi: https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings (there are two main prevailing styles - the Google one and numpydoc, which is a bit more verbose). For reference this is how this docstring might look:

```suggestion
    """Returns two flattened lists of Question--Answer indices given authors numbers to
    remove/retain, in the case where the forget target can be easily structured.
    Assumes equal questions per author, and sorted authors.

    Args:
        forgotten_author_numbers: List of integers containing author indexes about which
            facts should be removed.
        n_authors: Total number of authors.
        q_removal: Number of questions/facts that should be removed.
        q_per_author: Integer defining the number of questions to remove per author.

    Returns:
        A tuple (forget_indices, retain_indices) containing the indices of questions
        pertaining the authors in forgotten_author_numbers and the remaining question
        indices, respectively.
    """
```

Reply:
(and Copilot is pretty good at reducing the time taken to write docstrings btw)